### PR TITLE
[regression](compaction) Add case to test single replica compaction

### PR DESCRIPTION
**Pull request by Xiaoccer**

### CODE REVIEW

**Comment by Xiaoccer** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

File: regression-test/suites/compaction/test_single_replica_compaction.groovy [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        Thread.sleep(20000)
+        
+        // find the master be for single replica compaction
+        Boolean found = false
+        String master_backend_id;
+        List<String> follower_backend_id = new ArrayList<>()
+        String tablet_id = tablets[0][0]
+        for (String[] tablet in tablets) {
+            String trigger_backend_id = tablet[2]
+            def tablet_status = getTabletStatus(backendId_to_backendIP[trigger_backend_id], backendId_to_backendHttpPort[trigger_backend_id], tablet_id);
```
Comment (xiaokang) [inline regression-test/suites/compaction/test_single_replica_compaction.groovy:148]: why use the same tablet_id in the for loop?

Comment (xiaokang) [reply regression-test/suites/compaction/test_single_replica_compaction.groovy:148]: Are the tablets are different replica of the only 1 tablet?

Comment (xiaokang) [reply regression-test/suites/compaction/test_single_replica_compaction.groovy:148]: Add some comment.

File: be/src/olap/tablet.cpp [hunk 2]
```diff
@@ -1450,6 +1450,22 @@ void Tablet::get_compaction_status(std::string* json_result) {
                                            root.GetAllocator());
     root.AddMember("last base status", base_compaction_status_value, root.GetAllocator());
 
+    TReplicaInfo replica_info;
+    std::string dummp_token;
+    rapidjson::Value fetch_addr;
+    if (tablet_meta()->tablet_schema()->enable_single_replica_compaction() &&
+        StorageEngine::instance()->get_peer_replica_info(tablet_id(), &replica_info,
+                                                         &dummp_token)) {
+        std::string addr = replica_info.host;
```
Comment (xiaokang) [inline be/src/olap/tablet.cpp:1459]: std::string addr = replica_info.host + ":" + std::to_string(replica_info.brpc_port);

File: regression-test/suites/compaction/test_single_replica_compaction.groovy (hunk 1 above)
Comment (Xiaoccer) [reply regression-test/suites/compaction/test_single_replica_compaction.groovy:148]: > Are the tablets are different replica of the only 1 tablet?

yes, the table only has one bucket with 3 replication num

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by bobhan1 — APPROVED** [review]

File: regression-test/suites/compaction/test_single_replica_compaction.groovy [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                disableAutoCompaction = Boolean.parseBoolean(((List<String>) ele)[2])
+            }
+        }
+        set_be_config.call("disable_auto_compaction", "true")
+        has_update_be_config = true
+
+        def triggerCompaction = { be_host, be_http_port, compact_type, tablet_id ->
+            StringBuilder sb = new StringBuilder();
+            sb.append("curl -X POST http://${be_host}:${be_http_port}")
+            sb.append("/api/compaction/run?tablet_id=")
```
Comment (csun5285) [inline regression-test/suites/compaction/test_single_replica_compaction.groovy:62]: It does not employ single replica compaction. use table_id =

**Pull request merged**